libs/@blockprotocol/type-system/crate/src/ontology/entity_type/repr.rs

```diff
@@ -52,6 +56,54 @@ pub struct EntityType {
     links: repr::Links,
 }
 
+impl EntityType {
+    /// Merges another entity type into this one.
+    ///
+    /// This will:
+    ///   - remove the other entity type from the `allOf`
+    ///   - merge the `properties` and `required` fields
+    ///   - merge the `links` field
+    ///
+    /// # Notes
+    ///
+    /// - This does not validate the resulting entity type.
+    /// - The `required` field may have a different order after merging.
```

Comment:
Follow-up from above. I like this wording better :)

Reply:
The "This will:" section was mainly written by Copilot, it's better than me 🙃